crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -520,15 +520,18 @@ impl Dispatch<xdg_toplevel::XdgToplevel, ()> for WaylandClientState {
         if let xdg_toplevel::Event::Configure {
             width,
             height,
-            states: _states,
+            states,
         } = event
         {
-            if width == 0 || height == 0 {
-                return;
-            }
+            // todo!(linux): according to the protocol, if either of these are 0,
+            // "the client should decide its own window dimension", what should it be?
+            let width = if width != 0 { width } else { 500 };
```

Comment:
It seems like we should just not do anything in this case, right?

Reply:
In the case where exactly one of them is non-zero we still need to resize, should I add `WaylandWindowState::resize_width` and `WaylandWindowState::resize_height` methods or change `WaylandWindowState::resize` to take `Option<i32>`? Also do you know if there's a reason they are `i32` and not `u32` or even `NonZeroU32`?

Reply:
I feel like the wording in the spec is really poor around this:

> If the width or height arguments are zero, it means the client should decide its own window **dimension**

Because "dimension" is not plural I'm pretty certain it talks about each dimension individually.

There are niche cases where I think this can actually matter, like in scrolling compositors (e.g. [niri](https://github.com/YaLTeR/niri)), I think changing the available height for the row of windows will send configure events with the `width` being `0` and it shouldn't be ignored or the window will be cut off or squished.

I found an [example](https://github.com/linebender/druid/blob/f45c892dc52bda7a1713429585d540675b317fea/druid-shell/src/backend/wayland/surfaces/toplevel.rs#L93-L100) in [`druid`](https://github.com/linebender/druid/tree/master), I think they are handling it wrong in exactly this way, but otherwise they do just not resize, it seems like the most reasonable thing to do.

Reply:
Druid's solution is good enough IMO. When niri user's start filing issues, we'll fix it.

Reply:
I don't see a reason to make the code intentionally wrong and not follow the protocol.
I reworked the resizing functions to take `Option<NonZeroU32>` (which is the same representation as a `u32` and size as an `i32`), the bounds themselves are `Bounds<u32>` and not `Bounds<NonZeroU32>` because `NonZeroU32` isn't `Default`, and I didn't want to touch more code than I already have.